### PR TITLE
Fixes for DragonFlyBSD

### PR DESCRIPTION
**Pull request by mneumann**

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -565,7 +565,12 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
     r = sendfile(in_fd, out_fd, req->off, &len, NULL, 0);
 #endif
 
-    if (r != -1 || len != 0) {
+    /*
+     * Do NOT rely on the assumption that the value of 'len' is not being
+     * modified by the operating system during the system call in case of an
+     * error, unless explicitly specified for the error (EAGAIN, EINTR).
+     */
```
Comment (bnoordhuis) [inline src/unix/fs.c]: LGTM but I'd just state that EAGAIN and EINTR update `len`.  Double negatives are harder to parse.

**Comment by saghul** [comment]
@mneumann ping?

**Comment by mneumann** [comment]
@saghul: Sorry for the delay. @bnoordhuis : Your implicit assumption is that _ONLY_ EAGAIN or EINTR will ever update ```len```. At least according to the documentation, the value of ```len``` is only valid in case of EGAIN and EINTR. For example ```sendfile()``` might return an error code (```r == -1```) while setting the ```len``` to an arbitrary (invalid) value if ```errno != EAGAIN```. I think this must have happend on DragonFly in a certain case, which is ok according to the man page.

**Comment by saghul** [comment]
@mneumann No worries! FWIW, the wording did confuse me a bit. Would it be correct to say: "`len` is only updated to a meaningful value in case of error for EAGAIN or EINTR" ?

**Comment by mneumann** [comment]
@saghul Yes, exactly. For the other error conditions, ```len``` could be an arbitrary value.

**Comment by saghul** [comment]
@mneumann Got it. Mind updating the comment to that then? :-)

**Comment by mneumann** [comment]
@saghul : better? :)

**Comment by saghul** [comment]
LGTM, thanks!

**Comment by mneumann** [comment]
Thank you :)

**Comment by saghul** [comment]
Landed in d58e94408fb433fc8138131a9732e6c48510a5be and 8b90faff6c55cc4bfbc19f0aec655b89fa7283be thanks!

**Comment by mneumann** [comment]
Thanks!

**Pull request closed without merging**